src/cast_core/src/StreamStatsReporter.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cast_core/actors/StreamStatsReporter.hpp>
+
+#include <memory>
+
+#include <yaml-cpp/yaml.h>
+
+#include <bsoncxx/array/view.hpp>
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/json.hpp>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/database.hpp>
+
+#include <boost/log/trivial.hpp>
+#include <boost/throw_exception.hpp>
+
+#include <gennylib/Cast.hpp>
+#include <gennylib/MongoException.hpp>
+#include <gennylib/context.hpp>
+
+#include <value_generators/DocumentGenerator.hpp>
+
+
+namespace genny::actor {
+
+struct StreamStatsReporter::PhaseConfig {
+    mongocxx::database database;
+    TimeSpec pollInterval;
+    bsoncxx::document::value metricsCommand = bsoncxx::from_json("{}");
+
+    PhaseConfig(PhaseContext& phaseContext, mongocxx::database&& db, ActorId id)
+        : database{db},
+          pollInterval{phaseContext["PollInterval"].to<TimeSpec>()} {
+        int threads = phaseContext.actor()["Threads"].to<int>();
+        assert(threads == 1);
+
+        std::string streamProcessorName = phaseContext["StreamProcessorName"].to<std::string>();
+        bsoncxx::builder::stream::document builder;
+        metricsCommand = builder << "streams_getMetrics" << "" << bsoncxx::builder::stream::finalize;
```

Comment:
so using `getMetrics` rather than `getStats` only because `getMetrics` is more "real time" and `getStats` is only updated periodically (which seems to be very infrequent since the in-memory operator processes everything thats queued up in one run iteration, which could be many many thousand documents)